File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/models/SharedPebbleLink.kt
package app.pbbls.android.features.path.models

/**
 * The public share-by-link URL for a pebble (M51). Canonical host (invite-link
 * convention, see ConnectionsService's INVITE_HOST); the uuid is the
 * capability, lowercased to match how the web app prints ids.
 */
object SharedPebbleLink {
    fun url(pebbleId: String): String = "https://www.pbbls.app/p/${pebbleId.lowercase()}"
}

File: apps/ios/Pebbles/Features/Path/Models/SharedPebbleLink.swift
import Foundation

/// The public share-by-link URL for a pebble (M51). Canonical host, matching
/// the invite-link convention (Connection.swift) — the uuid is the capability,
/// lowercased to match how the web app prints ids.
enum SharedPebbleLink {
    static func url(for pebbleId: UUID) -> URL {
        URL(string: "https://www.pbbls.app/p/\(pebbleId.uuidString.lowercased())")!
    }
}
